domain/operation/types.go

```diff
@@ -2,3 +2,105 @@
 // Licensed under the AGPLv3, see LICENCE file for details.
 
 package operation
+
+import (
+	"time"
+
+	"github.com/juju/juju/core/machine"
+	"github.com/juju/juju/core/unit"
+)
+
+// QueryParams represents the parameters used for querying operations.
+type QueryParams struct {
+	Target
+	ActionNames []string
+	Status      []string
+
+	// These attributes are used to support client side
+	// batching of results.
+	Limit  *int
+	Offset *int
+}
+
+// QueryResult represents the result of a query operation.
+type QueryResult struct {
+	Operations []OperationInfo
+	Truncated  bool
+}
+
+// OperationInfo represents the information about an operation.
+type OperationInfo struct {
+	OperationID string
+	Summary     string
+	Fail        string
+	Enqueued    time.Time
+	Started     time.Time
+	Completed   time.Time
+	Status      string
+	Machines    []MachineTaskResult
+	Units       []UnitTaskResult
+	Truncated   bool
+	Error       error
+}
+
+// RunParams represents the parameters used for running operations.
+type RunParams struct {
+	Target
+	TaskParams
+}
+
+// TaskParams represents the parameters used for running tasks.
+type TaskParams struct {
```

Comment:
todo: Replace `params` with `args` in these structures. It makes the facade code more difficult to read.  `params `isn't usually used in outside of rpc interactions. We usually use `args`.